talisker/request_id.py

```diff
@@ -66,21 +66,27 @@ def decorator(id_func):
     def wrapper(func):
         @wraps(func)
         def decorator(*args, **kwargs):
+            current_id = get()
             id = id_func(*args, **kwargs)
+            if current_id == id:
```

Comment:
I'm not really sure what this does. Perhaps a comment explaining why it short-circuits the func, i.e. does not set/restore the id, would be a good idea?